test-explanation.md

```diff
@@ -0,0 +1,56 @@
+
+#### Supply Policy
+The Supply Policy Contract has a single external function, also called `rebase()`, this is not to be confused with the rebase method in the Ampleforth ERC-20 Contract. This `rebase()` method is publicly callable by anyone, but will only execute at most once every 24 hours. Opening this method up, helps to remove us as a necessary central party in the system’s execution. If we fail to call `rebase()` for any reason, others are free to make that call in our place. The rebase() method first queries the Market Oracle to get the current price. If the price is within `priceThreshold` of the target price, no supply policy change is applied. Otherwise, the absolute `supplyDelta` is equal to `(price-target)*totalSupply/target`. For example, if Amples are trading for $1.15, the absolute `totalSupply` increase will be 15%. Next, it applies a “rebase reaction lag” to dampen the supply change. At launch, the reaction lag will be 30 days. Finally, the Ampleforth ERC-20 token is instructed to adjust its supply by the dampened value. Continuing with the example above, the dampened increase would be (15% / 30 days) = 0.5% per day. Due to the unpredictability of when transactions get mined into a block, and because at least 24 hours must pass before a rebase executes, there will always be slightly more than 24 hours between rebases. This means that, even though our rebase time is 24hrs, the rebase call will “drift” slightly over time. Based on our measurements on Ethereum’s Rinkeby testnet, we expect this drift to be about 1 hour per year. So if rebase calls execute at 0:00 UTC time on Jan 1st, they would execute 1:00 UTC a year later. A public countdown timer to the next allowable rebase operation each day is displayed on the Ampleforth Protocol dashboard.
+
+#### Market Oracle
+The Market Oracle Contract provides data from the outside world to be used by the Supply Policy Contract. Specifically, it returns the 24-hour volume-weighted Ample Price from exchanges. At launch, oracle will have a trusted whitelist of sources and the price is calculated as the median of the sources.
+1. Only whitelisted addresses can provide market data.
+2. A market report must exist on-chain publicly for at least 1 hour before it can be used by the supply policy.
+3. A market report will expire on-chain if a new report is not provided before 6 hours elapse.
+
+
+## Ampleforth Fork Deployment
+
+| Component                 | Address                                           |
+|---------------------------|---------------------------------------------------|
+| Deployer                  | 0x08150bAB13edC834FD5b436C9416dC849f410C66       |
+| UFragments deployed to    | 0x0Df6D221DbB5cabAc86777E9FF6B455a1b2cFAD4       |
+| Implementation            | 0x86B1c1eD8E4161eAbE7c78D182ef16b9b49c9146       |
+| Market oracle to          | 0x15a98688AB4274C578d19874D5Cf3B2991d08A31       |
+| CPI oracle to             | 0x6b06Eb8f79457bC036d1a50e4bDcbeBD7AdD50c4       |
```

Comment:
and these oracles are they also part of AMPL and to be forkable with the same license or how does this work?

Reply:
bc we need to use a custom data feed most likely then

Reply:
is everything within the same repository

Reply:
ok cool stuff, the better we understand the oracle part the better! Since that will be the part we will need to change!